contrib/inventory/ec2.py

```diff
@@ -338,8 +338,8 @@ def read_settings(self):
         else:
             self.all_elasticache_nodes = False
 
-        # boto configuration profile (prefer CLI argument)
-        self.boto_profile = self.args.boto_profile
+        # boto configuration profile (prefer environment variables then CLI argument)
+        self.boto_profile = os.environ.get('AWS_PROFILE', self.args.boto_profile)
```

Comment:
I had made the exact change you suggest but did `git commit --amend` rather than `git commit --amend -a` and so just pushed an amended commit message, unhelpfully.